controllers/registry.go

```diff
@@ -783,3 +790,15 @@ func (r *DPAReconciler) populateAzureRegistrySecret(bsl *velerov1.BackupStorageL
 
 	return nil
 }
+
+func (r *DPAReconciler) verifySecretContent(secretName string, secretKey string) error {
+	secret, err := r.getProviderSecret(secretName)
+	if err != nil {
+		return err
+	}
+	data, foundKey := secret.Data[secretKey]
+	if !foundKey || len(data) == 0 {
```

Comment:
@kaovilai this is the check you wanted for the bug you are working on Azure?

Reply:
Not quite.. the azure bug from meeting for imagestream is related to decoded data at secret key, not the length of secret key itself.

Reply:
alright, got it now 

so, solution would be adding a next step here, that would differ from azure, aws and gcp, and checking the content of `data` (not only its length)?

this can be a separate PR

Reply:
> so, solution would be adding a next step here, that would differ from azure, aws and gcp, and checking the content of `data` (not only its length)?
> 
> this can be a separate PR

Tracked in https://issues.redhat.com/browse/OADP-3562

comment there as you see fit.